plugins/node/opentelemetry-instrumentation-graphql/src/instrumentation.ts

```diff
@@ -58,11 +58,14 @@ const DEFAULT_CONFIG: GraphQLInstrumentationConfig = {
   mergeItems: false,
   depth: -1,
   allowValues: false,
+  ignoreFields: [],
 };
 
 const supportedVersions = ['>=14'];
 
 export class GraphQLInstrumentation extends InstrumentationBase {
+  private _ignoreMap!: Map<string, boolean>;
```

Comment:
@obecny - you wrote
> hash map is faster the set

I guess you refer to this issue?
https://stackoverflow.com/questions/49841972/javascript-how-come-map-has-is-so-much-faster-than-set-has-and-array-indexof

I don't know the current status of the issue, [but from looking on node source code](https://github.com/nodejs/node/blob/36805e852471cc7a24524a329848da5c8f163864/deps/v8/src/compiler/js-call-reducer.cc#L7557), it seems that `Set.has` is nowdays treated the same as `Map.has` in V8:
```cpp
Reduction JSCallReducer::ReduceMapPrototypeHas(Node* node) {
  return ReduceCollectionPrototypeHas(node, CollectionKind::kMap);
}

Reduction JSCallReducer::ReduceSetPrototypeHas(Node* node) {
  return ReduceCollectionPrototypeHas(node, CollectionKind::kSet);
}
```

I think for our case this "ignoreFields" config list is expected to be very small, so anyhow I see little value in harming instrumentation readability for this optimization.
Adding a `Map` from string to `true` just seems wired to me.

Reply:
Originally it was using ".includes" so if you optimise something it is best to use the fastest way imho. This might be called thousands of times during usage so I would say it is still better to use less cpu than more. This is a kind of thing you do once and then you probably wont be touching this again for a long time. 
```
ignoreFields" config list is expected to be very small
```
what if not ?, besides that the list can be small but still it can be called many times per second.